### DISCUSSION TITLE
Navigating to files in TEXINPUTS

### DISCUSSION THREAD

**Discussion by dpo** [latex-lsp/texlab#1228] [Q&A]
Hi,

I use the texlab LSP to work on LaTeX projects in LunarVim (a distribution of Neovim). I often need to navigate to style files and other files that are `\include`d or `\usepackage`d in documents. If I hit `gd` on the file name, LunarVim seamlessly navigates to the file IF that file is in the same git repo. However, it doesn't nagivate to other files, even if they are in locations listed in my `TEXINPUTS`. Is that a LunarVim limitation or a Texlab bug/missing feature?

Thank you.

**Comment by dpo** [comment]
Just to add to my question, the LunarVim folks feel that this isn't a LunarVim limitation. And just to be clear, `gd` is the Neovim shortcut for "go to definition".

**Comment by lnay** [comment]
I looked through issue discussions about this not long ago. I believe `texlab` is unaffected by `TEXINPUTS` (confirmed by experiments). If everything you need from outside the workspace/repo can be used as packages in preambles, then you can put them in your user texmf tree. Note that this will involving running some command to rehash some file, not just moving files.

**Reply by dpo** [reply]
Sorry, I don’t see how that’s related to my question. I'm not talking about LaTeX being able to discover files in my `TEXINPUTS`, but about Texlab letting me navigate to them from one of my `.tex` files.

**Reply by lnay** [reply]
On top of the latex compiling correctly, texlab also does go-to-definition correctly when the other file is in your texmf tree. That was just a suggestion to consider, the actual answer to your question was "it's a texlab limitation".

**Reply by dpo** [reply]
I see. Are there any plans to add such a feature in the future?

**Reply by pfoerster** [reply]
With https://github.com/latex-lsp/texlab/pull/1252, the server will have support for `TEXINPUTS` similar to the way it is done for `BIBINPUTS` currently.
